src/linker/Linker.Steps/LoadReferencesStep.cs

```diff
@@ -32,20 +32,42 @@
 
 namespace Mono.Linker.Steps
 {
-	public class LoadReferencesStep : BaseStep
+	public class LoadReferencesStep : IAssemblyStep
 	{
+		LinkContext _context;
+
+		LinkContext Context => _context;
+
 		readonly HashSet<AssemblyNameDefinition> references = new HashSet<AssemblyNameDefinition> ();
 
-		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		readonly HashSet<AssemblyDefinition> newReferences = new HashSet<AssemblyDefinition> ();
+
+		public void Initialize (LinkContext context)
+		{
+			_context = context;
+		}
+
+		public virtual void ProcessAssemblies (HashSet<AssemblyDefinition> assemblies)
```

Comment:
Why does it take HashSet? It seems like huge complication if any step could mutate that set

Reply:
This is going away with the new approach.